srcs/builtin/_util_commands.c

```diff
@@ -0,0 +1,43 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   _util_commands.c                                   :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: kitsuki <kitsuki@student.42tokyo.jp>       +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/05/22 02:05:53 by kitsuki           #+#    #+#             */
+/*   Updated: 2023/05/22 02:05:53 by kitsuki          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+size_t	get_strslen(char **src)
+{
+	size_t	count;
+
+	if (src == NULL)
+		return (0);
+	count = 0;
+	while (src[count] != NULL)
+		count++;
+	return (count);
+}
+
+bool	free_all(char ***dst)
+{
+	char	**tmp;
+
+	if (dst == NULL || *dst == NULL)
```

Comment:
`dst`が not NULL かつ `*dst`が NULL の場合に、`dst`がfreeされずリークになります。

通常時
```
[
	"abc",
	"def",
	NULL
]
```
として、リーク発生ケースが `[NULL]` ですね。

```suggestion
	if (dst == NULL)
```

`*dst`のNULLチェックは35行目のwhile文でも行なっているので、単に`*dst == NULL`を削除するだけで問題なく動作するかなと思います。

Reply:
いえ、この関数は、文字列配列(char **)のポインタを取る関数です。
元の変数にNULLを代入するためにこうしています。
そのため、*dst も NULL なのは本来ないべきです。
35行目のNULLチェックは、**dstです。(tmp = *dst で、whileの条件は *tmp != NULL のため)

Reply:
三重ポインタを見落としてました… ありがとうございます！